src/test/java/formflow/library/pdf/PDFFormFillerTest.java

```diff
@@ -0,0 +1,106 @@
+package formflow.library.pdf;
+
+import com.lowagie.text.pdf.AcroFields;
+import com.lowagie.text.pdf.PdfReader;
+import com.lowagie.text.pdf.parser.PdfTextExtractor;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatCode;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ActiveProfiles("test")
+@SpringBootTest(properties = {"form-flow.path=flows-config/test-flow.yaml"})
+class PDFFormFillerTest {
+
+  private final PDFFormFiller pdfFormFiller = new PDFFormFiller();
+  private final String pdf = "/pdfs/testPdf.pdf";
+
+  @Test
+  void shouldMapAllFieldTypes() throws IOException {
+    String textFieldValue = "Michael";
+    String radioValue = "option2";
+    String checkboxSelectedValue = "Yes";
+    String unselectedCheckboxValue = "Off";
+
+    Collection<PdfField> fields = List.of(
+            new PdfField("TEXT_FIELD", textFieldValue),
+            new PdfField("RADIO_BUTTON", radioValue),
+            new PdfField("CHECKBOX_OPTION_1", checkboxSelectedValue),
+            new PdfField("CHECKBOX_OPTION_3", checkboxSelectedValue)
+    );
+
+    File file = pdfFormFiller.fill(pdf, fields, false);
+
+    assertEquals(getField(file, "TEXT_FIELD"), textFieldValue);
+    assertEquals(getField(file, "RADIO_BUTTON"), radioValue);
+    assertEquals(getField(file, "CHECKBOX_OPTION_1"), checkboxSelectedValue);
+    assertEquals(getField(file, "CHECKBOX_OPTION_2"), unselectedCheckboxValue);
+    assertEquals(getField(file, "CHECKBOX_OPTION_3"), checkboxSelectedValue);
+  }
+
+  @Test
+  void shouldAccept_a_with_macron() throws IOException {
+    String textFieldValue = "Michaelā";
+    Collection<PdfField> fields = List.of(
+            new PdfField("TEXT_FIELD", textFieldValue)
+    );
+
+    File file = pdfFormFiller.fill(pdf, fields, true);
+
+    assertTrue(getText(file).contains("Michael ā")); // openpdf's text extraction introduces a space
```

Comment:
That's odd. Does it do this with all diacritics? 

In general, I think it'd be great to test a list of diacritics here- mainly the ones used in Spanish and some of the Vietnamese ones.  I'd throw in an umlaut as well.

Reply:
It is just a result of imperfect text extraction from PDF files. The PDF displays "Michaelā" correctly (no embedded space). It is not limited to odd characters - it happens for all kinds of text, even regular English words.

Reply:
Confirmed @bseeger, I paired with Vraj on this and the space isn't caused by the special character, spaces can happen anywhere and is more the result of `getText()`. It's not a perfect mirror of how the PDF looks.

Reply:
Is it hard to add more diacritics into this?   Just a quick run through of the more common ones?

Reply:
We can add more diacritics, just being clear that the diacritics is not the cause of the weird space.

Reply:
I added another [test for more diacritics](https://github.com/codeforamerica/form-flow/pull/454/files#diff-b2c6d5ebf543755e0887ef2cda8797a3f291cfafbc5c4a0ee9f54fed7936d040R58)